test/irb/test_input_method.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: false
+
+require "test/unit"
+require "irb"
+
+module TestIRB
+  class TestRelineInputMethod < Test::Unit::TestCase
+    def setup
+      @conf_backup = IRB.conf.dup
+      IRB.conf[:LC_MESSAGES] = IRB::Locale.new
+    end
+
+    def teardown
+      IRB.conf.replace(@conf_backup)
+    end
+
+    def test_initialization
+      IRB::RelineInputMethod.new
+
+      assert_nil Reline.completion_append_character
+      assert_equal '', Reline.completer_quote_characters
+      assert_equal IRB::InputCompletor::BASIC_WORD_BREAK_CHARACTERS, Reline.basic_word_break_characters
+      assert_equal IRB::InputCompletor::CompletionProc, Reline.completion_proc
+      assert_equal IRB::InputCompletor::PerfectMatchedProc, Reline.dig_perfect_match_proc
+    end
+
+    def test_initialization_without_use_autocomplete
+      original_show_doc_proc = Reline.dialog_proc(:show_doc)&.dialog_proc
+      empty_proc = Proc.new {}
+      Reline.add_dialog_proc(:show_doc, empty_proc)
+
+      IRB.conf[:USE_AUTOCOMPLETE] = false
+
+      IRB::RelineInputMethod.new
+
+      refute Reline.autocompletion
+      assert_equal empty_proc, Reline.dialog_proc(:show_doc).dialog_proc
+    ensure
+      Reline.add_dialog_proc(:show_doc, original_show_doc_proc, Reline::DEFAULT_DIALOG_CONTEXT)
+    end
+
+    def test_initialization_with_use_autocomplete
+      original_show_doc_proc = Reline.dialog_proc(:show_doc)&.dialog_proc
+      empty_proc = Proc.new {}
+      Reline.add_dialog_proc(:show_doc, empty_proc)
+
+      IRB.conf[:USE_AUTOCOMPLETE] = true
+
+      IRB::RelineInputMethod.new
+
+      assert Reline.autocompletion
+      assert_equal IRB::RelineInputMethod::SHOW_DOC_DIALOG, Reline.dialog_proc(:show_doc).dialog_proc
+    ensure
+      Reline.add_dialog_proc(:show_doc, original_show_doc_proc, Reline::DEFAULT_DIALOG_CONTEXT)
+    end
+
+    def test_initialization_with_use_autocomplete_but_without_rdoc
+      original_show_doc_proc = Reline.dialog_proc(:show_doc)&.dialog_proc
+      empty_proc = Proc.new {}
+      Reline.add_dialog_proc(:show_doc, empty_proc)
+
+      IRB.conf[:USE_AUTOCOMPLETE] = true
+
+      without_rdoc do
+        IRB::RelineInputMethod.new
+      end
+
+      assert Reline.autocompletion
+      # doesn't register show_doc dialog
+      assert_equal empty_proc, Reline.dialog_proc(:show_doc).dialog_proc
+    ensure
+      Reline.add_dialog_proc(:show_doc, original_show_doc_proc, Reline::DEFAULT_DIALOG_CONTEXT)
+    end
+
+    def without_rdoc(&block)
```

Comment:
How about moving the `without_rdoc` method to the common file `test/lib/helper.rb`, as the logic: `require 'rdoc'` and `resure LoadError` is in other parts too?

```
$ grep -r rdoc lib/
lib/irb/cmd/help.rb:        require 'rdoc/ri/driver'
lib/irb/completion.rb:        require 'rdoc'
lib/irb/easter-egg.rb:          require "rdoc"
lib/irb/input-method.rb:          require 'rdoc'
```

Reply:
I want to leave it to the next PR (which I'll open after this is merged) because the goal of this one is to move out of the `require "rdoc"`. Increasing the coverage is just a good addition. WDYT?

Reply:
Yes, I agree with you. It can be the next PR. Without the `test_initialization_with_use_autocomplete_but_without_rdoc` test in this PR, it's still better than the current situation.